File: Solutions/Implement_Rand10()_Using_Rand7()/Implement_Rand10()_Using_Rand7().cpp
class Solution {
public:
    int n, m;
    int rand10() {
        // initialising the values
		n = rand7(), m = 7;
        // making sure n is in the right 1-5 range and, if we can, we settle m too
		while (n > 5) {
            m = n - 5;
            n = rand7();
        }
        // if m was not settled, we do it here
		while (m == 7) m = rand7();
        return (m % 2 ? 5 : 0) + n;
    }
};

//Most common
class Solution {
public:
    int rand10() {
        int col, row, idx;
        do{
            col=rand7();
            row=rand7();
            idx=col+(row-1)*7;
        }while(idx<40);
        
        return (idx-40)+1;
            
    }
};

//Fastest
class Solution {
public:
    int rand10() {
        int x = 7;
        while (x >= 7) { x = rand7(); }

        int y = 6;
        while (y >= 6) { y = rand7(); }
        
        int ans = 0;
        if (x > 3) ans = 5;
        ans += y;
        return ans;
    }
};

File: Solutions/Implement_Rand10()_Using_Rand7()/Implement_Rand10()_Using_Rand7().js
var rand10 = function() {
  let i = 7;
  let j = 6;
  while (i > 6) i = rand7(); // i could be 1, 2, 3, 4, 5, or 6
  while (j > 5) j = rand7(); // j could be 1, 2, 3, 4, or 5
  if (i % 2 === 0) { // if i end up being even (50% the chance because it could be 2, 4, 6)
    return j; // return 1, 2, 3, 4, or 5
  } else { //if i end up being odd (also 50% of the chance because it could be 1, 3, 5)
    return j + 5; // return 6, 7, 8, 9, or 10
  }
};

//Fastest
var rand10 = function() {
    
    let idx = -1;
    
    while (true) {
        
        const row = rand7();
        const col = rand7();
        
        idx = col + (row - 1) * 7;
        
        if (idx <= 40) {
            return 1 + (idx - 1) % 10;
        }
    }
};